packages/block-editor/src/components/block-list/hover-area.js

```diff
@@ -1,82 +1,41 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
-import { withSelect } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
 
-class HoverArea extends Component {
-	constructor() {
-		super( ...arguments );
-		this.state = {
-			hoverArea: null,
-		};
-		this.onMouseLeave = this.onMouseLeave.bind( this );
-		this.onMouseMove = this.onMouseMove.bind( this );
-	}
-
-	componentWillUnmount() {
-		if ( this.props.container ) {
-			this.toggleListeners( this.props.container, false );
-		}
-	}
-
-	componentDidMount() {
-		if ( this.props.container ) {
-			this.toggleListeners( this.props.container );
-		}
-	}
-
-	componentDidUpdate( prevProps ) {
-		if ( prevProps.container === this.props.container ) {
-			return;
-		}
-		if ( prevProps.container ) {
-			this.toggleListeners( prevProps.container, false );
-		}
-		if ( this.props.container ) {
-			this.toggleListeners( this.props.container, true );
-		}
-	}
+const useHoveredArea = ( wrapper ) => {
+	const [ hoveredArea, setHoveredArea ] = useState( null );
 
-	toggleListeners( container, shouldListnerToEvents = true ) {
-		const method = shouldListnerToEvents ? 'addEventListener' : 'removeEventListener';
-		container[ method ]( 'mousemove', this.onMouseMove );
-		container[ method ]( 'mouseleave', this.onMouseLeave );
-	}
-
-	onMouseLeave() {
-		if ( this.state.hoverArea ) {
-			this.setState( { hoverArea: null } );
-		}
-	}
+	useEffect( () => {
+		const onMouseLeave = () => {
+			if ( hoveredArea ) {
+				setHoveredArea( null );
+			}
+		};
 
-	onMouseMove( event ) {
-		const { isRTL, container } = this.props;
-		const { width, left, right } = container.getBoundingClientRect();
+		const onMouseMove = ( event ) => {
+			const { width, left, right } = wrapper.current.getBoundingClientRect();
 
-		let hoverArea = null;
-		if ( ( event.clientX - left ) < width / 3 ) {
-			hoverArea = isRTL ? 'right' : 'left';
-		} else if ( ( right - event.clientX ) < width / 3 ) {
-			hoverArea = isRTL ? 'left' : 'right';
-		}
+			let newHoveredArea = null;
+			if ( ( event.clientX - left ) < width / 3 ) {
```

Comment:
Is it possible to have `width / 3` extracted into a variable? this way it would be easier to reason about this at first sight.